### ISSUE TITLE
All of my accounts are ending with error

### ISSUE THREAD

**Issue by deadman36g** [sealemar/BingRewards#169]
All my accounts finishing with the following

The server couldn't fulfill the request.
Error code:  302

Mobile searches seem to be ok but it is the PC searches that are not completing

**Comment by amayer5125** [comment]
Are you sure you are running the newest version? Do you have any other error output?

**Comment by deadman36g** [comment]
Just updated to make sure I was running the newest but problem still exists. I have noticed that one the BR dashboard is states for PC searched to start with 'US News' could this be the problem and if so is there a fix?

**Comment by amayer5125** [comment]
In your config file what query generator are you using?

**Comment by deadman36g** [comment]
<configuration>
    <general
        betweenQueriesInterval="15.0"
        betweenQueriesSalt="12.0"
        betweenAccountsInterval="300"
        betweenAccountsSalt="600"
        openTopLinkRange="5"
        openLinkChance=".7"
        addSearchesDesktop="0"
        addSearchesDesktopSalt="20"
        addSearchesMobile="0"
        addSearchesMobileSalt="20" />

**Comment by deadman36g** [comment]
oops, sorry wrong info

 <queries generator="wikipedia" />
</configuration>

**Comment by deadman36g** [comment]
<queries generator="wikipedia" />
</configuration>

**Comment by deadman36g** [comment]
I tried changing the 'wikipedia' to 'bing' but now I get the following when running the script

Traceback (most recent call last):
  File "main.py", line 304, in <module>
    EventsProcessor.onScriptFailure(config, e)
  File "main.py", line 302, in <module>
    __run(config)
  File "main.py", line 227, in __run
    reportItem = __processAccountUserAgent(config, account, bingCommon.USER_AGENTS_PC, doSleep)
  File "main.py", line 214, in __processAccountUserAgent
    __processAccount(config, httpHeaders, agents, reportItem, account.password)
  File "main.py", line 183, in __processAccount
    earnRewards(config, httpHeaders, userAgents, reportItem, accountPassword)
  File "main.py", line 60, in earnRewards
    results = bingRewards.process(rewards, verbose)
  File "pkg/bingRewards.py", line 403, in process
    res = self.__processSearch(r, verbose)
  File "pkg/bingRewards.py", line 287, in __processSearch
    queries = queryGenerator.generateQueries(searchesCount, history)
  File "pkg/queryGenerators/bing.py", line 157, in generateQueries
    s = newsPage.index(startMarker)
ValueError: substring not found

**Comment by amayer5125** [comment]
No it should say wikipedia. That is fine. I just wanted to make sure it wasnt a problem with that.

**Comment by ekauzlar** [comment]
Also getting this error, newest version, stopped working as of today.

I added a print statement to print out the full error.

The server couldn't fulfill the request.
Error code:  302
HTTP Error 302: Found - Redirection to url 'microsoft-edge:https://www.bing.com/rewards/dashboard' is not allowed

**Comment by Cryson** [comment]
I am also getting the error 302 mentioned above. Python version is 2.7.

**Comment by cloudlyfe-aldwin** [comment]
Syntax error in http_error_302

https://github.com/cloudlyfe-aldwin/BingRewards/commit/63a06ce9cd8e7b99541b8745574de82603e9f404

**Comment by cloudlyfe-aldwin** [comment]
The "Earn more credits" on the flyout page is the culprit. Certain accounts have links with microsoft-edge protocol, which would kick off Edge browser on Windows 10 when clicked. If your browser doesn't support the protocol it'll just redirect you back to the dashboard page. The script is complaining about  302 redirect.

There are 2 ways to fix the issue.
1. Update bingFlyoutParser.py to handle "Earn more credits" as Action.INFORM
2. HTTPErrorProcessor in bingRewards.py to ignore 302 error.

```
class HTTPErrorProcessor(urllib2.HTTPErrorProcessor):
    def http_response(self, request, response):
        code, msg, hdrs = response.code, response.msg, response.info()

        # Stop 302 redirection.
        if code == 302: return response

        if not (200 <= code < 300):
            response = self.parent.error('http', request, response, code, msg, hdrs)
        return response

    https_response = http_response
```

Pull request https://github.com/sealemar/BingRewards/pull/173 fixes the issue utilizing the 1st approach. Although the HttpErrorProcessor solution would be a better and long term solution, I opted to take the simpler approach.

**Comment by wipedout** [comment]
Question (novice) on updating the FlyoutParser:
Do I add another line to the list of Aliases, such as this:
EARN_MORE_CREDITS        = (13,    "Earn More Credits",                     None, False, Action.INFORM)

and then add EARN_MORE_CREDITS to the list of "ALL = ( [alias list]  ) just below that defined list?

Thanks for confirming, or posting a replacement FlyoutParser.

**Comment by amayer5125** [comment]
@wipedout this appears to be unrelated to this issue. Can you start a new issue so the discussion is in its own thread?

**Comment by amayer5125** [comment]
This should be fixed by #192
